### DISCUSSION TITLE
Is there one way to set Block's base_point later after created

### DISCUSSION THREAD

**Discussion by hom** [mozman/ezdxf#734] [Q&A] [answered]
Hi, I want to set Block's base_point attribute after it is created, such as `doc.blocks.new(name="BLOCKNAME")` or `doc.new_anonymous_block()`
```python
blk = doc.new_anonymous_block()
```
I try to use ways below, and always return `AttributeError: can't set attribute 'base_point` error message.
```python
blk.base_point = Vec3(-10, -10, 0)
```
and
```python
blk.__setattr__("base_point", Vec3(-10, -10, 0))
```
and
```python
setattr(blk, "base_point", Vec3(-10, -10, 0))
```
and some other ways

This is my source code:
```python
import ezdxf
from ezdxf.math import Vec3
from ezdxf.enums import TextEntityAlignment
from ezdxf.tools.text_size import text_size

doc = ezdxf.new(dxfversion="R2018", setup=["linetypes"])
msp = doc.modelspace()
blk = doc.blocks.new_anonymous_block()
text = blk.add_text(text="0.1", dxfattribs={"height": 4, "width": 1, }).set_placement((10, 10), align=TextEntityAlignment.MIDDLE)
blk.__setattr__("base_point", Vec3((-10, -10, 0)))
msp.add_blockref(blk.name, insert=(0, 0))

doc.saveas("output/set_block_base_point.dxf")
```

Anyone here can help me? Thanks very much:)

**Comment by mozman** [comment, marked as answer]
The method `new_anonymous_block()`  has an argument `base_point` to define the block base point.

If that's not good enough, the method returns a `BlockLayout` instance that has an attribute `block` which is a `Block` instance of the related `BLOCK` entity, which represents that `BlockLayout`. The `Block` entity contains the desired DXF attribute `base_point`.

```Python

blk.block.dxf.base_point = Vec3(-10, -10, 0)
```

**Reply by hom** [reply]
Yeah, thanks very much, it's amazing.

The error is what I change the `base_point` attribute in returned `new_anonymous_block()` directly, which try to set `base_point` to `BlockLayout` entity, so it set failed. And I need to get the `Block` entity first when changing `base_point` later. Got it!!

Thanks @mozman again:)
